### PR TITLE
genai: add File API

### PR DESCRIPTION
**Pull request by jba**
Add support for the file API, which allows uploading large files
to the service so they can be used to generate content.

The GetFile, DeleteFile and ListFiles RPCs are straightforward.
We use the same GAPIC client for them as we have been using for the
rest of the API.

However, the CreateFile RPC involves a media upload, which GAPICs
do not support. We rely instead on the discovery client, which
we have to generate ourselves. That involves a shell script
to do the work, and a copy of an internal package (gensupport)
from the repo that contains the discovery clients.

### CODE REVIEW

File: genai/internal/generativelanguage/v1beta/generativelanguage-gen.go
Comment (jba) [inline genai/internal/generativelanguage/v1beta/generativelanguage-gen.go:1]: No need to review; generated code.

File: genai/internal/gensupport/buffer.go
Comment (jba) [inline genai/internal/gensupport/buffer.go:1]: No need to review anything in this directory; copied code.

File: genai/internal/version.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package internal
+
+// Version is the current tagged release of the library.
+const Version = "0.172.0"
```
Comment (eliben) [inline genai/internal/version.go:8]: It's confusing to have two `internal.Version` meaning different things in the code - is this a necessary structure, or can these version files be placed in packages?

**Review by eliben — APPROVED** [review]
yikes

File: genai/internal/version.go (hunk 1 above)
Comment (jba) [reply genai/internal/version.go:8]: I got rid of this one.

File: genai/client.go [hunk 2]
```diff
@@ -29,17 +29,22 @@ import (
...
+	gld "github.com/google/generative-ai-go/genai/internal/generativelanguage/v1beta" // discovery client
 
-	"github.com/google/generative-ai-go/internal"
 	"github.com/google/generative-ai-go/internal/support"
 	"google.golang.org/api/iterator"
 	"google.golang.org/api/option"
 )
 
+// version is the current tagged release of this package.
+const version = "v0.11.0"
```
Comment (eliben) [inline genai/client.go:40]: There are still two version constants. The one in `genai/internal/version` is different from this one

Comment (jba) [reply genai/client.go:40]: Oh, good point. There's only one now.

Comment (jba) [reply genai/client.go:40]: Oh dang, I already merged it. See forthcoming PR.

Comment (0994115340street-lang) [reply genai/client.go:40]: > There are still two version constants. The one in `genai/internal/version` is different from this one

**Pull request merged**